src/txmempool.h

```diff
@@ -684,6 +684,8 @@ class CTxMemPool
         return (mapTx.count(gtxid.GetHash()) != 0);
     }
 
+    std::optional<CTxMemPoolEntry::CTxMemPoolEntryRef> GetEntry(const uint256& hash) const LIFETIMEBOUND;
```

Comment:
Concept ACK to handing out references to the entry instead of iterators into mapTx

ISTM that this should require holding `CTxMemPool::cs`?

Reply:
Yeah, missed that. Will add.